src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1382,31 +1682,56 @@ private Event nextValue_1_1() {
                         break;
                     }
                 }
-                event = super.nextValue();
-                if (parent == null && isPositionedOnEncodingDirective()) {
+                if (isEvaluatingEExpression) {
+                    evaluateNext();
+                } else {
+                    event = super.nextValue();
+                }
+                if (minorVersion == 1 && parent == null && isPositionedOnEncodingDirective()) {
                     encodingDirectiveReader.resetState();
                     state = State.ON_ION_ENCODING_SEXP;
                     continue;
                 }
                 break;
             }
+        } else if (isEvaluatingEExpression) {
+            evaluateNext();
         } else {
             event = super.nextValue();
         }
         if (valueTid != null && valueTid.isMacroInvocation) {
-            // TODO delegate to macroEvaluatorIonReader while this invocation is active.
-            throw new UnsupportedOperationException("Cannot yet invoke a macro.");
+            if (macroEvaluator == null) {
+                // If the macro evaluator is null, it means there is no active macro table. Do not attempt evaluation,
+                // but allow the user to do a raw read of the parameters if this is a core-level reader.
```

Comment:
When could this scenario occur and/or be useful? It's impossible to perform a raw read of the parameters if you don't know the macro signatures, right?

Reply:
I added this for testing (so that the core reader tests didn't need to set up a macro context). If you implicitly know the macro signature, then you can navigate the reader correctly through the stream. I added a TODO to evaluate whether this will ever be useful outside of tests. If it won't, then we can just change the tests and remove this.